src/media.cpp

```diff
@@ -296,7 +298,7 @@ void FeBaseStream::push_packet( AVPacket *pkt )
 
 void FeBaseStream::free_packet( AVPacket *pkt )
 {
-	av_free_packet( pkt );
+	av_packet_unref( pkt );
```

Comment:
This switch to av_packet_unref fails for me since the linux distro I am on uses an older version of FFmpeg/libav.  Could you wrap this with a version check macro like is done elsewhere, so it only uses the newer function with newer FFmpeg versions?  Looking at https://github.com/FFmpeg/FFmpeg/blob/master/doc/APIchanges it looks like something like this would work:

```
#if (LIBAVCODEC_VERSION_INT >= AV_VERSION_INT( 57, 12, 0 ))
        av_packet_unref( pkt );
#else
       av_free_packet( pkt );
#endif